New types added to TinyJson

fatihyldrim opened the pull request:
TinyJson does not work when object has a DateTime and Guid. The type conversions added to TinyJson. Also tests are modified.

lugehorsam commented:
Nice contribution @fatihyldrim  -- I wanted to check if there was a reason why you couldn't serialize this as a `string`?

fatihyldrim commented:
Thank you @lugehorsam -- My entities and DTOs contain datetime and guid types to provide ease of operations. And I added these types for serialize and deserialize directly from DTOs. 

I added another test function to be clearer. Let me know if there is more efficient way.

lugehorsam commented:
👍
